db/crud_test.go

```diff
@@ -133,7 +137,10 @@ func TestRevisionStorageConflictAndTombstones(t *testing.T) {
 	rev3b_body := Body{}
 	rev3b_body["version"] = "3b"
 	rev3b_body[BodyDeleted] = true
-	_, err = db.PutExistingRevWithBody("doc1", rev3b_body, []string{"3-b", "2-b"}, false)
+	doc, newRev, err = db.PutExistingRevWithBody("doc1", rev3b_body, []string{"3-b", "2-b"}, false)
+	rev3b_body[BodyId] = doc.ID
```

Comment:
Would be nice to have the test not rely on these being in the body. I think it's mostly down to the `deepEquals` assertions?

Maybe we could get away with checking for the version/_deleted fields instead?

Reply:
I think in general this makes sense for future but it would take quite a while to move the current tests to use this format. If you think its worth it I'll go ahead.

Reply:
Yup, that's fair enough. It seemed a bit painful to refactor how tests work as part of this.